tasks.go

```diff
@@ -116,28 +116,20 @@ func NewError(err error) Task {
 	return func() error { return err }
 }
 
-type ResourceMatchFactory func(project, resource, substr string) ([]string, error)
-
 // GetNagiosTasks sends tasks to dump Nagios data for each project that contain
-// a Nagios pod. This function will output an error to the user if no Nagios pods
-// were found in any projects.
-func GetNagiosTasks(tasks chan<- Task, projects []string, basepath string, resourceFactory ResourceMatchFactory) {
+// a Nagios pod. It is an error if no projects contain a Nagios pod.
+func GetNagiosTasks(tasks chan<- Task, runner Runner, projects []string) {
 	foundANagiosPod := false
 	for _, p := range projects {
-		pods, err := resourceFactory(p, "pod", "nagios")
+		pods, err := getResourceNamesBySubstr(p, "pod", "nagios")
```

Comment:
Since it is not tested, I went with simplifying the code.
Eventually `getResourceNamesBySubstr` should take the runner as argument.

Then using a fake runner we can test the whole thing without having to inject a lot of different functions.